In [2]:
import numpy as np
import pandas as pd
from functions import *

In [3]:
df = pd.read_csv("data/returns.csv")
df.head()

Date  Return
0  2014-01-01      34
1  2014-02-01      34
2  2014-03-01      49
3  2014-04-01       2
4  2014-05-01      26

In [7]:
# Standard deviation
df["Return"].std()

np.float64(26.96979195083174)

Std interpret: data is quite spreaded, most of values are within the range of +-27 from the mean value.

In [23]:
# Give the first quartile:

print("Q1: " + str(df["Return"].quantile(0.25)))

# Give the empirical value of VaR(0.05), i.e., the fifth percentile of returns:
var = df["Return"].quantile(0.05)
print("VaR 0.05: " + str(var))

# Give the empirical value of ES(0.05), i.e., the average return for observations below the fifth percentile
es = df["Return"].loc[df["Return"] < var].mean()
print("ES 0.05: " + str(es))

Q1: -6.25
VaR 0.05: -29.05
ES 0.05: -35.833333333333336


Q1: 25% of returns are lower than -6.25 p.p.
Value at Risk: there is 5% chance that i can have lower returns than 29.5 p.p.
Expected shortfall: during the worst 5% months (when returns < -29.5), the expected average loss is -35.83 p.p.

In [24]:
# mean and test for the mean
mean = np.mean(df["Return"])
mean

np.float64(11.708333333333334)

in our sample mean return is 11.7%

In [25]:
import scipy.stats as stats

test_result = stats.ttest_1samp(df["Return"], popmean=9, alternative="greater")

print(test_result)

TtestResult(statistic=np.float64(1.1466268401990842), pvalue=np.float64(0.12691809617691754), df=np.int64(119))


there is no enough evidence that mean return is greater than 9. 